locales/i18n.js

```diff
@@ -19,11 +19,20 @@ function formatLocale(lang) {
 function urlOverrideLang(path) {
   var localPath = path || location.pathname;
   var localeCode = localPath.split('/')[1];
-  var pathname = localPath.split('/')[2] || localeCode;
+  localeCode = (/^[a-z]{2,3}(-[A-Z]{2})?$/.test(localeCode)) ? localeCode : null;
```

Comment:
Are you familiar with node and npm?

If so we have a little module that's a list of known langs we can check against.

https://www.npmjs.com/package/langmap

You can see an example of me using this in another site here: https://github.com/mozilla/advocacy.mozilla.org/commit/cbb5421190d29b9396f39fe84e32eaa2d6bb86c8#diff-bc8b2010218e61c4795f2d4928be206bR3